thirdeye-server/src/main/java/ai/startree/thirdeye/auth/ResourceIdentifier.java

```diff
@@ -15,37 +15,50 @@
 
 import ai.startree.thirdeye.spi.datalayer.dto.AbstractDTO;
 import ai.startree.thirdeye.spi.datalayer.dto.AlertDTO;
+import ai.startree.thirdeye.spi.datalayer.dto.AlertTemplateDTO;
+import javax.annotation.Nullable;
 
 public class ResourceIdentifier {
 
   static public final String DefaultNamespace = "default";
 
   public final String name;
   public final String namespace;
-  public final String entityType;
+  public final EntityType entityType;
 
-  public ResourceIdentifier(final String name, final String namespace, final String entityType) {
+  public ResourceIdentifier(final String name, @Nullable final String namespace,
+      final EntityType entityType) {
     this.name = name;
-    this.namespace = namespace;
+    this.namespace = namespace != null ? namespace : DefaultNamespace;
```

Comment:
nit: use optional()
```
optional(namespace).orElse(DEFAULT_NAMESPACE)
```

why? it looks cute :)

Reply:
Omg I looked for something like this haha! Ya much nicer.